pkg/cache/snapshot.go

```diff
@@ -128,6 +129,7 @@ func (c *ClusterQueue) snapshot() *ClusterQueue {
 		AllocatableResourceGeneration: c.AllocatableResourceGeneration,
 		Usage:                         make(FlavorResourceQuantities, len(c.Usage)),
 		Workloads:                     make(map[string]*workload.Info, len(c.Workloads)),
+		GuaranteedQuota:               make(map[kueue.ResourceFlavorReference]map[corev1.ResourceName]int64, len(c.GuaranteedQuota)),
```

Comment:
```suggestion
```

Should we allocate after we check if the feature, `lendingLimit`  is enabled?

Reply:
Yes we do. Both in [L212](https://github.com/kubernetes-sigs/kueue/pull/1385/files#diff-441da8393964615cf5b020d0afb9199588e157e50718e83b527788eb679391ecR211) of pkg/cache/clusterqueue.go and [L150](https://github.com/kubernetes-sigs/kueue/pull/1385/files#diff-bd8411463d27f0b9477a855409c5133c437d8cae368eeff5c2e5957f26c23507R150) of pkg/cache/snashot.go, we have checked if the feature enabled.